symphony-bdk-legacy/symphony-api-client-java/src/main/java/internal/FileHelper.java

```diff
@@ -28,30 +28,33 @@
    */
   @SneakyThrows
   public static byte[] readFile(@Nonnull final String path) throws FileNotFoundException {
-
-    byte[] content;
-
-    if(!isClasspath(path) && new File(path).exists()) {
-      try(FileInputStream fis = new FileInputStream(path)) {
-          content = toByteArray(fis);
-          logger.debug("File loaded from system path : {}", path);
-        }
-        catch (Exception ex) {
-          throw new FileNotFoundException("Unable to load file from path : " + path);
-        }
-    }
-    else if (FileHelper.class.getResource(classpath(path)) != null) {
-      content = toByteArray(FileHelper.class.getResourceAsStream(classpath(path)));
+    if (!isClasspath(path) && new File(path).exists()) {
+      try (FileInputStream fis = new FileInputStream(path)) {
+        logger.debug("File loaded from system path : {}", path);
+        return toByteArray(fis);
+      } catch (Exception ex) {
+        throw new FileNotFoundException("Unable to load file from path : " + path);
+      }
+    } else if (FileHelper.class.getResource(classpath(path)) != null) {
```

Comment:
it is a bit weird that here we use getResource first and then getResourceAsStream (it was like that before)

also to load relative from this class is perhaps a bit weird for the user of the SDKs
perhaps we should just use ClassLoader.getSystemResourceAsStream ? (I started looking into ClassPathResource from Spring and it gets messy quickly, picking the right classloader, forward slash or not...)

Reply:
Yes indeed, in Spring things are messy.
But please check: https://docs.oracle.com/javase/8/docs/technotes/guides/lang/resources.html

> **Using Methods of java.lang.Class**
The getResource and getResourceAsStream methods find a resource with a given name. They return null if they do not find a resource with the specified name. The rules for searching for resources associated with a given class are implemented by the class's ClassLoader. The Class methods delegate to ClassLoader methods, after applying a naming convention: if the resource name starts with "/", it is used as is. Otherwise, the name of the package is prepended, after converting all periods (.) to slashes (/).

So for our usecases, I guess it is enough and will work if paths begin with `classpath:/`. Things get messier if there is no slash after `classpath:`.